Include hybrid values that are proven in trie proofs

hrxi opened the pull request:
This allows to use the proof to directly read values from the trie.

hrxi committed:
Include hybrid values that are proven in trie proofs

This allows to use the proof to directly read values from the trie.

sisou commented on primitives/src/trie/trie_proof_node.rs:
```diff
@@ -58,6 +59,29 @@ impl From<TrieNode> for TrieProofNode {
+}
+
+impl TrieProofNode {
+    pub fn into_value(self) -> Option<Vec<u8>> {
+        match self.value {
+            ProofValue::None => None,
+            ProofValue::LeafValue(value) => Some(value),
+            ProofValue::HybridHash(_) => None,
+            ProofValue::HybridValue(value) => Some(value),
+        }
+    }
```
Might it make more sense to not take self but use and return references?

```suggestion
    pub fn value(&self) -> Option<&Vec<u8>> {
        match &self.value {
            ProofValue::None => None,
            ProofValue::LeafValue(value) => Some(value),
            ProofValue::HybridHash(_) => None,
            ProofValue::HybridValue(value) => Some(value),
        }
    }
```

jsdanielh committed:
consensus: Use the new `verify_values` from `TrieProof`

- Use the new `verify_values` method from `TrieProof`.
- Remove the no longer needed `values` method in `TrieProof`.
